M2/cmake/check-libraries.cmake

```diff
@@ -40,7 +40,16 @@ endif()
 
 find_package(Threads	REQUIRED QUIET)
 find_package(LAPACK	REQUIRED QUIET)
-find_package(Boost	REQUIRED QUIET COMPONENTS regex ${Boost_stacktrace})
+find_package(Boost	REQUIRED QUIET COMPONENTS regex OPTIONAL_COMPONENTS stacktrace_backtrace stacktrace_addr2line)
+if(Boost_STACKTRACE_BACKTRACE_FOUND)
+  set(Boost_stacktrace_lib "Boost::stacktrace_backtrace")
+elseif(Boost_STACKTRACE_ADDR2LINE_FOUND)
+  set(Boost_stacktrace_lib "Boost::stacktrace_addr2line")
+else()
+  #fallback to header only mode
+  set(Boost_stacktrace_header_only YES)
```

Comment:
In this case you're assuming that the headers are found, but they may not be, right? Ideally we should get an error here if the headers aren't found rather than at the end when building M2-binary. You can do this with [`check_include_file`](https://cmake.org/cmake/help/latest/module/CheckIncludeFile.html).

Reply:
This is not relevant for macOS or distributions where Boost is one package, but for instance on Ubuntu boost-regex and boost-stacktrace are separate packages, and the latter may be missing.

Reply:
I can certainly add that check, but I don't know if any distro installs only half the header files. In particular on ubuntu libboost-stacktrace-dev includes only the libraries themselves, and libboost-dev seems to include all the headers.

See [https://packages.ubuntu.com/groovy/amd64/libboost-stacktrace1.71-dev/filelist](https://packages.ubuntu.com/groovy/amd64/libboost-stacktrace1.71-dev/filelist) and [https://packages.ubuntu.com/groovy/amd64/libboost1.71-dev/filelist](https://packages.ubuntu.com/groovy/amd64/libboost1.71-dev/filelist)

Additionally, I believe the autotools build checks that boost is at least version 1.65, which is the first version boost-stacktrace is avaliable, but doesn't directly check for the header.

Reply:
I see, you're right, it doesn't matter then.